### DISCUSSION TITLE
overfitting in 03_pytorch_computer_vision

### DISCUSSION THREAD

**Discussion by HamidHamedi** [mrdbourke/pytorch-deep-learning#672] [General]
Hello!

Right before 7. Building a CNN model, it is written in that "it seems like our model is overfitting on the training data".

But how we could find there is overfitting by comparing model_1_results and model_0_results?
As far as I see, model_1_results shows loss and accuracy for the test data, but as far as I know to conclude if there is overfitting we need also to look at loss in train data. Am I right?

**Comment by ghowelldsp** [comment]
I have the same question here.

I am assuming we are saying that the model is **overfitting** due to the training accuracy being slightly higher that the test accuracy. This happens in both models, so are we saying that both models are overfitting? However, it is only a marginal difference in both cases so not clear on how much greater the training accuracy would have to be to be classed as overfitting.

Thanks in advance!

EDIT: From a little research, I believe my assumption about the training accuracy being higher than the testing accuracy to be correct. However, given such a marginal increase in this particular case, I don't believe that the loss in accuracy achieved in the non-linear model can be attributed to overfitting in this case.

**Model 0: Linear**
<img width="655" alt="Screenshot 2025-06-07 at 22 22 30" src="https://github.com/user-attachments/assets/a709d375-7feb-4d1e-9277-78ea62a617d1" />

**Model 1: Non-linear**
<img width="426" alt="Screenshot 2025-06-07 at 22 24 02" src="https://github.com/user-attachments/assets/e19995a2-7fa9-4908-b4a6-e255cbdb4a08" />

**Comment by LuluW8071** [comment]
@HamidHamedi  Yes, loss metrics are generally used to evaluate whether the model is learning effectively on both training and test data. And, you should compare loss metrics after 3-5 epochs (except during finetuning or transfer learning) as its not reliable to compare in early epochs as model is starting learn and it is just randomly guessing. 
 
> Try using loss metrics as primary metric comparision and then other metrics like acc, precision, f1_score, ... to check if model is learning or not, then see if it is overfitting, underfitting or best model.

![image](https://github.com/user-attachments/assets/0d44b389-0889-4dfc-a363-e2f7791ce7a8)
